<img src="Logo_IT_ACademy.jpg" alt="Texto alternativo" style="float:right; width:200px;">


# Análisis del parque automovilístico de Barcelona.

## Introducción

Barcelona tiene más 800.000 vehículos censados, entre turismos, furgonetas, camiones, motos, ciclomotores y otros vehículos. El análisis del parque automovilístico de la ciudad, pude ayudar a la planificación de esta, desde la definición y dotación de zonas de zona regulada hasta la definición y dotación de zonas de carga y descarga, por ejemplo.

Por otro lado, puede ayudar a determinar el estado del parque automovilístico en términos de envejecimiento y/o en términos de contaminación mediante la clasificación por etiqueta ambiental.

Este análisis del parque automovilístico de la ciudad de Barcelona, se realizará utilizando los datos públicos del Ayuntamiento de Barcelona ( https://opendata-ajuntament.barcelona.cat/es/). 

## Objetivo

En el siguiente tiene como objetivo analizar la evolución de la totalidad de vehículos en los distintos distritos de Barcelona. Así mismo, tratar de sacar conclusiones en función de la cantidad de tipos vehículos que puedan a llegar a concentrarse en las distintos Barrios o Distritos de Barcelona.

Por otro lado, se realiza un modelo de predicción que permita proyectar la cantidad de vehículos por tipo y por Barrio y/o Distrito con el objetivo que pueda servir para planificar las distintas zonas de la ciudad, en términos de movilidad.

De los dos data sets obtenidos del servicio Open Data del Ayuntamiento de Barcelona se realizan los siguientes análisis:

   - Evolución del parque automovilístico de Barcelona (periodo 2019 – 2021)
   - Análisis del parque automovilístico por Distritos (periodo 2019 – 2021)
        - Tipo de vehículo
        - Antigüedad
        - Distintivo ambiental
   - Modelo de predicción de la evolución del parque automovilístico por zonas.


<div class="alert alert-block alert-info"><font size="5"><center><b>Análisis del parque automovilístico de Barcelona</u></b></left></font>

<div class="alert alert-block alert-info"><font size="5"><center><b><u>Carga de Datos</u></b></center></font>
    
$%$       
\begin{align*}Ricard\:Simonneau\end{align*}

\begin{align*}Data\hspace{2mm}Science\hspace{2mm}Student\ - IT\hspace{2mm}Academy\end{align*}

\begin{align*}Date : June\hspace{2mm}2023\end{align*}</div>

### Preparación de los datos

Para el desarrollo del proyecto, cargaremos los datos obtenidos en el Servicio de datos abiertos del Ajuntament de Barcelona: https://opendata-ajuntament.barcelona.cat/es/. Descargaremos los datasets siguientes:

-	Padrón de vehículos. Antigüedad por tipo de vehículo del parque de vehículos de la ciudad de Barcelona.
-	Tipología del parque de vehículos de la ciudad de Barcelona según distintivo ambiental.

Para la generación y preparación de los data frames utilizaremos la biblioteca PANDAS.


In [1]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.


In [2]:
import pandas as pd

#### Dataframes (Antigüedad):

- Padrón de vehículos: __Antigüedad por tipo de vehículo del parque de vehículos de la ciudad de Barcelona__

    - 2019
    - 2020
    - 2021
    - Dataframe total

#### Datos 2019
Carga del archivo '2019_antiguitat_tipus_vehicle' correspondiene al 'Padrón de vehículos. Antigüedad por tipo de vehículo del parque de vehículos de la ciudad de Barcelona' en 2019: número de vehículos por tipo, antigüedad, distrito y barrio.

In [3]:
# Carga de archivo '2019_antiguitat_tipus_vehicle.csv'

pd.read_csv('2019_antiguitat_tipus_vehicle.csv')

Antigüedad_tipo_2019 = pd.read_csv('2019_antiguitat_tipus_vehicle.csv')

pd.options.display.max_columns = None

Antigüedad_tipo_2019


Any Codi_Districte Nom_Districte Codi_Barri  Nom_Barri Seccio_Censal  \
0      2019              1  Ciutat Vella          1   el Raval             1   
1      2019              1  Ciutat Vella          1   el Raval             1   
2      2019              1  Ciutat Vella          1   el Raval             1   
3      2019              1  Ciutat Vella          1   el Raval             1   
4      2019              1  Ciutat Vella          1   el Raval             1   
...     ...            ...           ...        ...        ...           ...   
52086  2019             NC     No consta         NC  No consta            NC   
52087  2019             NC     No consta         NC  No consta            NC   
52088  2019             NC     No consta         NC  No consta            NC   
52089  2019             NC     No consta         NC  No consta            NC   
52090  2019             NC     No consta         NC  No consta            NC   

        Tipus_Vehicles                   Antiguitat  Nombre  
0             Turismes  Menys d'un any d'antiguitat      19  
1             Turismes                        1 any      10  
2             Turismes                       2 anys      19  
3             Turismes                       3 anys      17  
4             Turismes                       4 anys      19  
...                ...                          ...     ...  
52086       Furgonetes               Més de 20 anys       8  
52087          Camions               D'11 a 20 anys       2  
52088          Camions               Més de 20 anys       8  
52089  Altres vehicles               D'11 a 20 anys       1  
52090  Altres vehicles               Més de 20 anys       2  

[52091 rows x 9 columns]

In [4]:
# Estructura del DF

Antigüedad_tipo_2019.shape

(52091, 9)

In [5]:
# Obtenemos el total de vehículos censados en 2019

total_vehiculos_2019 = Antigüedad_tipo_2019['Nombre'].sum()
print("Total vehículos Barcelona - 2019:", total_vehiculos_2019)


Total vehículos Barcelona - 2019: 824801


In [6]:
# Revisamos en que variables tenemos valores 'null', 'NC' o 'No consta'

# Comprobar valores nulos
valores_nulos_2019 = Antigüedad_tipo_2019.isnull().sum()
print("Valores nulos:")
print(valores_nulos_2019)

# Comprobar valores 'NC'
valores_nc_2019 = Antigüedad_tipo_2019[Antigüedad_tipo_2019 == 'NC'].count()
print("Valores 'NC':")
print(valores_nc_2019)

# Comprobar valores 'No consta'
valores_no_consta_2019 = Antigüedad_tipo_2019[Antigüedad_tipo_2019 == 'No consta'].count()
print("Valores 'No consta':")
print(valores_no_consta_2019)


Valores nulos:
Any               0
Codi_Districte    0
Nom_Districte     0
Codi_Barri        0
Nom_Barri         0
Seccio_Censal     0
Tipus_Vehicles    0
Antiguitat        0
Nombre            0
dtype: int64
Valores 'NC':
Any                0
Codi_Districte    33
Nom_Districte      0
Codi_Barri        33
Nom_Barri          0
Seccio_Censal     33
Tipus_Vehicles     0
Antiguitat         0
Nombre             0
dtype: int64
Valores 'No consta':
Any                 0
Codi_Districte      0
Nom_Districte      33
Codi_Barri          0
Nom_Barri          33
Seccio_Censal       0
Tipus_Vehicles      0
Antiguitat        158
Nombre              0
dtype: int64


In [7]:
# Comrpobamos el total de vehículos no clasificados en Distritos ('NC') y el total de vehículos no clasificados por antigüedad ('No consta'):
    # En este caso los valores 'NC' de 'Codi_Districte' coinciden con los valores 'No consta' de 'Nom_Districte',
    ## así como con los valores 'NC' de 'Codi_Barri' y 'No consta' de 'Nom_Barri'.


Total_vehiculos_NC_2019 = Antigüedad_tipo_2019.loc[Antigüedad_tipo_2019['Codi_Districte'] == 'NC', 'Nombre'].sum()
print("Total vehículos no clasificados en Distritos - 2019:", Total_vehiculos_NC_2019)

Total_vehiculos_No_Consta_2019 = Antigüedad_tipo_2019.loc[Antigüedad_tipo_2019['Antiguitat'] == 'No consta', 'Nombre'].sum()
print("Total vehículos no clasificados por Antigüedad - 2019:", Total_vehiculos_No_Consta_2019)


Total vehículos no clasificados en Distritos - 2019: 347
Total vehículos no clasificados por Antigüedad - 2019: 210


In [8]:
# Comprobamos si existe alguna coincidencia entre variables, es decir si alguna fila que contiene variable 'Codi_Districte' = 'NC'
## también contiene variable 'Antiguitat' = 'No consta'

Total_vehiculos_NC_2019_series = pd.Series([Total_vehiculos_NC_2019])
Total_vehiculos_No_Consta_2019_series = pd.Series([Total_vehiculos_No_Consta_2019])

coincidencia_2019 = Total_vehiculos_NC_2019_series.isin(Total_vehiculos_No_Consta_2019_series).any()

if coincidencia_2019:
    print("Hay coincidencia en los valores de total vehículos 'Nombre'.")
else:
    print("No hay coincidencia en los valores de total vehículos 'Nombre'.")

No hay coincidencia en los valores de total vehículos 'Nombre'.


#### Observaciones:

Tenemos un total de __824.801__ vehículos censados en Barcelona en 2019, de los cuales __347__ no están clasificados ni por distritos ni por barrios y __210__ no consta su antigüedad. Para el análisis del total del conjunto, teniendo en cuenta que estos 347 vehículos suponen menos de un 0,1% del total del parque automovílistico, los obviamos para no reclasificarlos aleatoriamente a ninguna zona de la ciudad ni con ningún tipo de antigüedad.

Al realziar la comprabación de 'coincindencia' aseguramos que los valores 'NC' de 'Codi_Districte' no coinciden con los valors 'No Consta' de la variable 'Antiguitat', es decir, ningun registro que tenga un valor 'NC' de 'Codi_Districte' tiene una valor 'No Consta' de  'Antiguitat'. Podemos asegurar que un total de __347__ vehículos no clasificados por 'Codi_Districte' y un total de __210__ no clasificados por 'Antiguitat', no coinciden en ningun registro y por tanto pueden obviarse para el análisis del DF.

In [9]:
# Eliminamos la filas que contengan 'Codi_Districte' = 'NC' i las filas que contengan 'Antiguitat' = 'No consta'

Antigüedad_tipo_2019_limpio = Antigüedad_tipo_2019.drop(Antigüedad_tipo_2019[(Antigüedad_tipo_2019['Codi_Districte'] == 'NC') | 
        (Antigüedad_tipo_2019['Antiguitat'] == 'No consta')].index)

Antigüedad_tipo_2019_limpio

Any Codi_Districte Nom_Districte Codi_Barri            Nom_Barri  \
0      2019              1  Ciutat Vella          1             el Raval   
1      2019              1  Ciutat Vella          1             el Raval   
2      2019              1  Ciutat Vella          1             el Raval   
3      2019              1  Ciutat Vella          1             el Raval   
4      2019              1  Ciutat Vella          1             el Raval   
...     ...            ...           ...        ...                  ...   
52053  2019             10    Sant Martí         73  la Verneda i la Pau   
52054  2019             10    Sant Martí         73  la Verneda i la Pau   
52055  2019             10    Sant Martí         73  la Verneda i la Pau   
52056  2019             10    Sant Martí         73  la Verneda i la Pau   
52057  2019             10    Sant Martí         73  la Verneda i la Pau   

      Seccio_Censal   Tipus_Vehicles                   Antiguitat  Nombre  
0                 1         Turismes  Menys d'un any d'antiguitat      19  
1                 1         Turismes                        1 any      10  
2                 1         Turismes                       2 anys      19  
3                 1         Turismes                       3 anys      17  
4                 1         Turismes                       4 anys      19  
...             ...              ...                          ...     ...  
52053           143          Camions               D'11 a 20 anys       8  
52054           143          Camions               Més de 20 anys       2  
52055           143  Altres vehicles                       4 anys       1  
52056           143  Altres vehicles               D'11 a 20 anys       5  
52057           143  Altres vehicles               Més de 20 anys       2  

[51900 rows x 9 columns]

In [10]:
# Comprobamos en el nuevo DF si existe algun valor 'null', 'NC' o 'No consta'

# Comprobar valores nulos
valores_nulos_2019_limpio = Antigüedad_tipo_2019_limpio.isnull().sum()
print("Valores nulos:")
print(valores_nulos_2019_limpio)

# Comprobar valores 'NC'
valores_nc_2019_limpio = Antigüedad_tipo_2019_limpio[Antigüedad_tipo_2019_limpio == 'NC'].count()
print("Valores 'NC':")
print(valores_nc_2019_limpio)

# Comprobar valores 'No consta'
valores_no_consta_2019_limpio = Antigüedad_tipo_2019_limpio[Antigüedad_tipo_2019_limpio == 'No consta'].count()
print("Valores 'No consta':")
print(valores_no_consta_2019_limpio)


Valores nulos:
Any               0
Codi_Districte    0
Nom_Districte     0
Codi_Barri        0
Nom_Barri         0
Seccio_Censal     0
Tipus_Vehicles    0
Antiguitat        0
Nombre            0
dtype: int64
Valores 'NC':
Any               0
Codi_Districte    0
Nom_Districte     0
Codi_Barri        0
Nom_Barri         0
Seccio_Censal     0
Tipus_Vehicles    0
Antiguitat        0
Nombre            0
dtype: int64
Valores 'No consta':
Any               0
Codi_Districte    0
Nom_Districte     0
Codi_Barri        0
Nom_Barri         0
Seccio_Censal     0
Tipus_Vehicles    0
Antiguitat        0
Nombre            0
dtype: int64


In [13]:
# Determinamos cuantos valores existen de cada variable:

for columna in Antigüedad_tipo_2019_limpio.columns:
    valores_unicos_2019 = Antigüedad_tipo_2019_limpio[columna].value_counts()
    cantidad_valores_unicos_2019 = len(valores_unicos_2019)
    print("Variable:", columna)
    print("Cantidad de valores únicos:", cantidad_valores_unicos_2019)
    print("Total de cada valor:")
    print(valores_unicos_2019)
    print()


Variable: Any
Cantidad de valores únicos: 1
Total de cada valor:
2019    51900
Name: Any, dtype: int64

Variable: Codi_Districte
Cantidad de valores únicos: 10
Total de cada valor:
2     8564
10    7318
7     5916
3     5623
8     5510
5     4816
9     4612
6     4045
4     2778
1     2718
Name: Codi_Districte, dtype: int64

Variable: Nom_Districte
Cantidad de valores únicos: 10
Total de cada valor:
Eixample              8564
Sant Martí            7318
Horta-Guinardó        5916
Sants-Montjuïc        5623
Nou Barris            5510
Sarrià-St. Gervasi    4816
Sant Andreu           4612
Gràcia                4045
Les Corts             2778
Ciutat Vella          2718
Name: Nom_Districte, dtype: int64

Variable: Codi_Barri
Cantidad de valores únicos: 73
Total de cada valor:
9     1866
60    1827
31    1679
19    1669
6     1618
      ... 
58      92
47      92
12      75
56      58
42      49
Name: Codi_Barri, Length: 73, dtype: int64

Variable: Nom_Barri
Cantidad de valores únicos: 73
Tot

--------------------------------------------------------------------------------------------------------------------------------

#### Datos 2020
Carga del archivo '2020_antiguitat_tipus_vehicle' correspondiene al 'Padrón de vehículos. Antigüedad por tipo de vehículo del parque de vehículos de la ciudad de Barcelona' en 2020: número de vehículos por tipo, antigüedad, distrito y barrio.

In [14]:
# Carga de archivo '2020_antiguitat_tipus_vehicle.csv'

pd.read_csv('2020_antiguitat_tipus_vehicle.csv')

Antigüedad_tipo_2020 = pd.read_csv('2020_antiguitat_tipus_vehicle.csv')

pd.options.display.max_columns = None

Antigüedad_tipo_2020

Any Codi_Districte Nom_Districte Codi_Barri  Nom_Barri Seccio_Censal  \
0      2020              1  Ciutat Vella          1   el Raval             1   
1      2020              1  Ciutat Vella          1   el Raval             1   
2      2020              1  Ciutat Vella          1   el Raval             1   
3      2020              1  Ciutat Vella          1   el Raval             1   
4      2020              1  Ciutat Vella          1   el Raval             1   
...     ...            ...           ...        ...        ...           ...   
52972  2020             NC     No consta         NC  No consta            NC   
52973  2020             NC     No consta         NC  No consta            NC   
52974  2020             NC     No consta         NC  No consta            NC   
52975  2020             NC     No consta         NC  No consta            NC   
52976  2020             NC     No consta         NC  No consta            NC   

        Tipus_Vehicles                   Antiguitat  Nombre  
0             Turismes  Menys d'un any d'antiguitat      11  
1             Turismes                        1 any      23  
2             Turismes                       2 anys      11  
3             Turismes                       3 anys      18  
4             Turismes                       4 anys      17  
...                ...                          ...     ...  
52972          Camions                        1 any       1  
52973          Camions               D'11 a 20 anys       1  
52974          Camions               Més de 20 anys       9  
52975  Altres vehicles               D'11 a 20 anys       3  
52976  Altres vehicles               Més de 20 anys       2  

[52977 rows x 9 columns]

In [15]:
# Estructura del DF

Antigüedad_tipo_2020.shape

(52977, 9)

In [16]:
# Obtenemos el total de vehículos censados

total_vehiculos_2020 = Antigüedad_tipo_2020['Nombre'].sum()
print("Total vehículos Barcelona - 2020:", total_vehiculos_2020)


Total vehículos Barcelona - 2020: 822211


In [17]:
# Revisamos en que variables tenemos valores 'null', 'NC' o 'No consta'

# Comprobar valores nulos
valores_nulos_2020 = Antigüedad_tipo_2020.isnull().sum()
print("Valores nulos:")
print(valores_nulos_2020)

# Comprobar valores 'NC'
valores_nc_2020 = Antigüedad_tipo_2020[Antigüedad_tipo_2020 == 'NC'].count()
print("Valores 'NC':")
print(valores_nc_2020)

# Comprobar valores 'No consta'
valores_no_consta_2020 = Antigüedad_tipo_2020[Antigüedad_tipo_2020 == 'No consta'].count()
print("Valores 'No consta':")
print(valores_no_consta_2020)


Valores nulos:
Any               0
Codi_Districte    0
Nom_Districte     0
Codi_Barri        0
Nom_Barri         0
Seccio_Censal     0
Tipus_Vehicles    0
Antiguitat        0
Nombre            0
dtype: int64
Valores 'NC':
Any                0
Codi_Districte    41
Nom_Districte      0
Codi_Barri        41
Nom_Barri          0
Seccio_Censal     41
Tipus_Vehicles     0
Antiguitat         0
Nombre             0
dtype: int64
Valores 'No consta':
Any                 0
Codi_Districte      0
Nom_Districte      41
Codi_Barri          0
Nom_Barri          41
Seccio_Censal       0
Tipus_Vehicles      0
Antiguitat        129
Nombre              0
dtype: int64


In [18]:
# Comrpobamos el total de vehículos no clasificados en Distritos ('NC') y el total de vehículos no clasificados por antigüedad ('No consta'):
    # En este coso los valores 'NC' de 'Codi_Districte' coinciden con los valores 'No consta' de 'Nom_Districte',
    ## así como con los valores 'NC' de 'Codi_Barri' y 'No consta' de 'Nom_Barri'.


Total_vehiculos_NC_2020 = Antigüedad_tipo_2020.loc[Antigüedad_tipo_2020['Codi_Districte'] == 'NC', 'Nombre'].sum()
print("Total vehículos no clasificados en Distritos - 2020:", Total_vehiculos_NC_2020)

Total_vehiculos_No_Consta_2020 = Antigüedad_tipo_2020.loc[Antigüedad_tipo_2020['Antiguitat'] == 'No consta', 'Nombre'].sum()
print("Total vehículos no clasificados por Antigüedad - 2020:", Total_vehiculos_No_Consta_2020)


Total vehículos no clasificados en Distritos - 2020: 415
Total vehículos no clasificados por Antigüedad - 2020: 158


In [19]:
# Comprobamos si existe alguna coincidencia entre variables, es decir si alguna fila que contiene variable 'Codi_Districte' = 'NC'
## también contiene variable 'Antiguitat' = 'No consta'

Total_vehiculos_NC_2020_series = pd.Series([Total_vehiculos_NC_2020])
Total_vehiculos_No_Consta_2020_series = pd.Series([Total_vehiculos_No_Consta_2020])

coincidencia_2020 = Total_vehiculos_NC_2020_series.isin(Total_vehiculos_No_Consta_2020_series).any()

if coincidencia_2020:
    print("Hay coincidencia en los valores de total vehículos 'Nombre'.")
else:
    print("No hay coincidencia en los valores de total vehículos 'Nombre'.")

No hay coincidencia en los valores de total vehículos 'Nombre'.


#### Observaciones:

Tenemos un total de __822.211__ vehículos censados en Barcelona en 2020, de los cuales __415__ no están clasificados ni por distritos ni por barrios y __158__ no consta su antigüedad. Para el análisis del total del conjunto, teniendo en cuenta que estos 573 vehículos suponen menos de un 0,1% del total del parque automovílistico, los obviamos para no reclasificarlos aleatoriamente a ninguna zona de la ciudad ni con ningún tipo de antigüedad.

Al realziar la comprabación de 'coincindencia' aseguramos que los valores 'NC' de 'Codi_Districte' no coinciden con los valors 'No Consta' de la variable 'Antiguitat', es decir, ningun registro que tenga un valor 'NC' de 'Codi_Districte' tiene una valor 'No Consta' de  'Antiguitat'. Podemos asegurar que un total de __415__ vehículos no clasificados por 'Codi_Districte' y un total de __158__ no clasificados por 'Antiguitat', no coinciden en ningun registro y por tanto pueden obviarse para el análisis del DF.

In [20]:
# Eliminamos la filas que contengan 'Codi_Districte' = 'NC' i las filas que contengan 'Antiguitat' = 'No consta'

Antigüedad_tipo_2020_limpio = Antigüedad_tipo_2020.drop(Antigüedad_tipo_2020[(Antigüedad_tipo_2020['Codi_Districte'] == 'NC') | 
        (Antigüedad_tipo_2020['Antiguitat'] == 'No consta')].index)

Antigüedad_tipo_2020_limpio

Any Codi_Districte Nom_Districte Codi_Barri            Nom_Barri  \
0      2020              1  Ciutat Vella          1             el Raval   
1      2020              1  Ciutat Vella          1             el Raval   
2      2020              1  Ciutat Vella          1             el Raval   
3      2020              1  Ciutat Vella          1             el Raval   
4      2020              1  Ciutat Vella          1             el Raval   
...     ...            ...           ...        ...                  ...   
52931  2020             10    Sant Martí         73  la Verneda i la Pau   
52932  2020             10    Sant Martí         73  la Verneda i la Pau   
52933  2020             10    Sant Martí         73  la Verneda i la Pau   
52934  2020             10    Sant Martí         73  la Verneda i la Pau   
52935  2020             10    Sant Martí         73  la Verneda i la Pau   

      Seccio_Censal   Tipus_Vehicles                   Antiguitat  Nombre  
0                 1         Turismes  Menys d'un any d'antiguitat      11  
1                 1         Turismes                        1 any      23  
2                 1         Turismes                       2 anys      11  
3                 1         Turismes                       3 anys      18  
4                 1         Turismes                       4 anys      17  
...             ...              ...                          ...     ...  
52931           143          Camions               D'11 a 20 anys      10  
52932           143          Camions               Més de 20 anys       2  
52933           143  Altres vehicles                       5 anys       1  
52934           143  Altres vehicles               D'11 a 20 anys       5  
52935           143  Altres vehicles               Més de 20 anys       2  

[52807 rows x 9 columns]

In [21]:
# Comprobamos en el nuevo DF si existe algun valor 'null', 'NC' o 'No consta'

# Comprobar valores nulos
valores_nulos_2020_limpio = Antigüedad_tipo_2020_limpio.isnull().sum()
print("Valores nulos:")
print(valores_nulos_2020_limpio)

# Comprobar valores 'NC'
valores_nc_2020_limpio = Antigüedad_tipo_2020_limpio[Antigüedad_tipo_2020_limpio == 'NC'].count()
print("Valores 'NC':")
print(valores_nc_2020_limpio)

# Comprobar valores 'No consta'
valores_no_consta_2020_limpio = Antigüedad_tipo_2020_limpio[Antigüedad_tipo_2020_limpio == 'No consta'].count()
print("Valores 'No consta':")
print(valores_no_consta_2020_limpio)


Valores nulos:
Any               0
Codi_Districte    0
Nom_Districte     0
Codi_Barri        0
Nom_Barri         0
Seccio_Censal     0
Tipus_Vehicles    0
Antiguitat        0
Nombre            0
dtype: int64
Valores 'NC':
Any               0
Codi_Districte    0
Nom_Districte     0
Codi_Barri        0
Nom_Barri         0
Seccio_Censal     0
Tipus_Vehicles    0
Antiguitat        0
Nombre            0
dtype: int64
Valores 'No consta':
Any               0
Codi_Districte    0
Nom_Districte     0
Codi_Barri        0
Nom_Barri         0
Seccio_Censal     0
Tipus_Vehicles    0
Antiguitat        0
Nombre            0
dtype: int64


In [22]:
# Determinamos cuantos valores existen de cada variable:

for columna in Antigüedad_tipo_2020_limpio.columns:
    valores_unicos_2020 = Antigüedad_tipo_2020_limpio[columna].value_counts()
    cantidad_valores_unicos_2020 = len(valores_unicos_2020)
    print("Variable:", columna)
    print("Cantidad de valores únicos:", cantidad_valores_unicos_2020)
    print("Nombres de cada valor:")
    print(valores_unicos_2020)
    print()

Variable: Any
Cantidad de valores únicos: 1
Nombres de cada valor:
2020    52807
Name: Any, dtype: int64

Variable: Codi_Districte
Cantidad de valores únicos: 10
Nombres de cada valor:
2     8664
10    7427
7     6012
3     5776
8     5633
5     4900
9     4691
6     4142
4     2808
1     2754
Name: Codi_Districte, dtype: int64

Variable: Nom_Districte
Cantidad de valores únicos: 10
Nombres de cada valor:
Eixample              8664
Sant Martí            7427
Horta-Guinardó        6012
Sants-Montjuïc        5776
Nou Barris            5633
Sarrià-St. Gervasi    4900
Sant Andreu           4691
Gràcia                4142
Les Corts             2808
Ciutat Vella          2754
Name: Nom_Districte, dtype: int64

Variable: Codi_Barri
Cantidad de valores únicos: 73
Nombres de cada valor:
9     1897
60    1850
31    1717
19    1690
6     1654
      ... 
47      93
58      92
12      76
56      57
42      48
Name: Codi_Barri, Length: 73, dtype: int64

Variable: Nom_Barri
Cantidad de valores únicos

--------------------------------------------------------------------------------------------------------------------------------

#### Datos 2021
Carga del archivo '2021_antiguitat_tipus_vehicle' correspondiene al 'Padrón de vehículos. Antigüedad por tipo de vehículo del parque de vehículos de la ciudad de Barcelona' en 2021: número de vehículos por tipo, antigüedad, distrito y barrio.

In [23]:
# Carga de archivo '2021_antiguitat_tipus_vehicle.csv'

pd.read_csv('2021_antiguitat_tipus_vehicle.csv')

Antigüedad_tipo_2021 = pd.read_csv('2021_antiguitat_tipus_vehicle.csv')

pd.options.display.max_columns = None

Antigüedad_tipo_2021

Any Codi_Districte Nom_Districte Codi_Barri  Nom_Barri Seccio_Censal  \
0      2021              1  Ciutat Vella          1   el Raval             1   
1      2021              1  Ciutat Vella          1   el Raval             1   
2      2021              1  Ciutat Vella          1   el Raval             1   
3      2021              1  Ciutat Vella          1   el Raval             1   
4      2021              1  Ciutat Vella          1   el Raval             1   
...     ...            ...           ...        ...        ...           ...   
52920  2021             NC     No consta         NC  No consta            NC   
52921  2021             NC     No consta         NC  No consta            NC   
52922  2021             NC     No consta         NC  No consta            NC   
52923  2021             NC     No consta         NC  No consta            NC   
52924  2021             NC     No consta         NC  No consta            NC   

        Tipus_Vehicles                   Antiguitat  Nombre  
0             Turismes  Menys d'un any d'antiguitat       8  
1             Turismes                        1 any      13  
2             Turismes                       2 anys      23  
3             Turismes                       3 anys      11  
4             Turismes                       4 anys      18  
...                ...                          ...     ...  
52920       Furgonetes               Més de 20 anys       9  
52921          Camions                       7 anys       1  
52922          Camions               Més de 20 anys       8  
52923  Altres vehicles               D'11 a 20 anys       2  
52924  Altres vehicles               Més de 20 anys       2  

[52925 rows x 9 columns]

In [24]:
# Estructura del DF

Antigüedad_tipo_2021.shape

(52925, 9)

In [25]:
# Obtenemos el total de vehículos censados en 2021

total_vehiculos_2021 = Antigüedad_tipo_2021['Nombre'].sum()
print("Total vehículos Barcelona - 2021:", total_vehiculos_2021)

Total vehículos Barcelona - 2021: 811673


In [26]:
# Revisamos en que variables tenemos valores 'null', 'NC' o 'No consta'

# Comprobar valores nulos
valores_nulos_2021 = Antigüedad_tipo_2021.isnull().sum()
print("Valores nulos:")
print(valores_nulos_2021)

# Comprobar valores 'NC'
valores_nc_2021 = Antigüedad_tipo_2021[Antigüedad_tipo_2021 == 'NC'].count()
print("Valores 'NC':")
print(valores_nc_2021)

# Comprobar valores 'No consta'
valores_no_consta_2021 = Antigüedad_tipo_2021[Antigüedad_tipo_2021 == 'No consta'].count()
print("Valores 'No consta':")
print(valores_no_consta_2021)

Valores nulos:
Any               0
Codi_Districte    0
Nom_Districte     0
Codi_Barri        0
Nom_Barri         0
Seccio_Censal     0
Tipus_Vehicles    0
Antiguitat        0
Nombre            0
dtype: int64
Valores 'NC':
Any                0
Codi_Districte    43
Nom_Districte      0
Codi_Barri        43
Nom_Barri          0
Seccio_Censal     43
Tipus_Vehicles     0
Antiguitat         0
Nombre             0
dtype: int64
Valores 'No consta':
Any                 0
Codi_Districte      0
Nom_Districte      43
Codi_Barri          0
Nom_Barri          43
Seccio_Censal       0
Tipus_Vehicles      0
Antiguitat        141
Nombre              0
dtype: int64


In [27]:
# Comrpobamos el total de vehículos no clasificados en Distritos ('NC') y el total de vehículos no clasificados por antigüedad ('No consta'):
    # En este coso los valores 'NC' de 'Codi_Districte' coinciden con los valores 'No consta' de 'Nom_Districte',
    ## así como con los valores 'NC' de 'Codi_Barri' y 'No consta' de 'Nom_Barri'.


Total_vehiculos_NC_2021 = Antigüedad_tipo_2021.loc[Antigüedad_tipo_2021['Codi_Districte'] == 'NC', 'Nombre'].sum()
print("Total vehículos no clasificados en Distritos - 2021:", Total_vehiculos_NC_2021)

Total_vehiculos_No_Consta_2021 = Antigüedad_tipo_2021.loc[Antigüedad_tipo_2021['Antiguitat'] == 'No consta', 'Nombre'].sum()
print("Total vehículos no clasificados por Antigüedad - 2021:", Total_vehiculos_No_Consta_2021)



Total vehículos no clasificados en Distritos - 2021: 394
Total vehículos no clasificados por Antigüedad - 2021: 179


In [28]:
# Comprobamos si existe alguna coincidencia entre variables, es decir si alguna fila que contiene variable 'Codi_Districte' = 'NC'
## también contiene variable 'Antiguitat' = 'No consta'

Total_vehiculos_NC_2021_series = pd.Series([Total_vehiculos_NC_2021])
Total_vehiculos_No_Consta_2021_series = pd.Series([Total_vehiculos_No_Consta_2021])

coincidencia_2021 = Total_vehiculos_NC_2021_series.isin(Total_vehiculos_No_Consta_2021_series).any()

if coincidencia_2021:
    print("Hay coincidencia en los valores de total vehículos 'Nombre'.")
else:
    print("No hay coincidencia en los valores de total vehículos 'Nombre'.")

No hay coincidencia en los valores de total vehículos 'Nombre'.


#### Observaciones:

Tenemos un total de __811.673__ vehículos censados en Barcelona en 2021, de los cuales __394__ no están clasificados ni por distritos ni por barrios y __179__ no consta su antigüedad. Para el análisis del total del conjunto, teniendo en cuenta que estos 573 vehículos suponen menos de un 0,1% del total del parque automovílistico, los obviamos para no reclasificarlos aleatoriamente a ninguna zona de la ciudad ni con ningún tipo de antigüedad.

Al realziar la comprabación de 'coincindencia' aseguramos que los valores 'NC' de 'Codi_Districte' no coinciden con los valors 'No Consta' de la variable 'Antiguitat', es decir, ningun registro que tenga un valor 'NC' de 'Codi_Districte' tiene una valor 'No Consta' de  'Antiguitat'. Podemos asegurar que un total de __394__ vehículos no clasificados por 'Codi_Districte' y un total de __179__ no clasificados por 'Antiguitat', no coinciden en ningun registro y por tanto pueden obviarse para el análisis del DF.

In [29]:
# Eliminamos la filas que contengan 'Codi_Districte' = 'NC' i las filas que contengan 'Antiguitat' = 'No consta'

Antigüedad_tipo_2021_limpio = Antigüedad_tipo_2021.drop(Antigüedad_tipo_2021[(Antigüedad_tipo_2021['Codi_Districte'] == 'NC') | 
        (Antigüedad_tipo_2021['Antiguitat'] == 'No consta')].index)

Antigüedad_tipo_2021_limpio

Any Codi_Districte Nom_Districte Codi_Barri            Nom_Barri  \
0      2021              1  Ciutat Vella          1             el Raval   
1      2021              1  Ciutat Vella          1             el Raval   
2      2021              1  Ciutat Vella          1             el Raval   
3      2021              1  Ciutat Vella          1             el Raval   
4      2021              1  Ciutat Vella          1             el Raval   
...     ...            ...           ...        ...                  ...   
52877  2021             10    Sant Martí         73  la Verneda i la Pau   
52878  2021             10    Sant Martí         73  la Verneda i la Pau   
52879  2021             10    Sant Martí         73  la Verneda i la Pau   
52880  2021             10    Sant Martí         73  la Verneda i la Pau   
52881  2021             10    Sant Martí         73  la Verneda i la Pau   

      Seccio_Censal   Tipus_Vehicles                   Antiguitat  Nombre  
0                 1         Turismes  Menys d'un any d'antiguitat       8  
1                 1         Turismes                        1 any      13  
2                 1         Turismes                       2 anys      23  
3                 1         Turismes                       3 anys      11  
4                 1         Turismes                       4 anys      18  
...             ...              ...                          ...     ...  
52877           143          Camions               Més de 20 anys       3  
52878           143  Altres vehicles  Menys d'un any d'antiguitat       1  
52879           143  Altres vehicles                       6 anys       1  
52880           143  Altres vehicles               D'11 a 20 anys       2  
52881           143  Altres vehicles               Més de 20 anys       3  

[52741 rows x 9 columns]

In [30]:
# Comprobamos en el nuevo DF si existe algun valor 'null', 'NC' o 'No consta'

# Comprobar valores nulos
valores_nulos_2021_limpio = Antigüedad_tipo_2021_limpio.isnull().sum()
print("Valores nulos:")
print(valores_nulos_2021_limpio)

# Comprobar valores 'NC'
valores_nc_2021_limpio = Antigüedad_tipo_2021_limpio[Antigüedad_tipo_2021_limpio == 'NC'].count()
print("Valores 'NC':")
print(valores_nc_2021_limpio)

# Comprobar valores 'No consta'
valores_no_consta_2021_limpio = Antigüedad_tipo_2021_limpio[Antigüedad_tipo_2021_limpio == 'No consta'].count()
print("Valores 'No consta':")
print(valores_no_consta_2021_limpio)

Valores nulos:
Any               0
Codi_Districte    0
Nom_Districte     0
Codi_Barri        0
Nom_Barri         0
Seccio_Censal     0
Tipus_Vehicles    0
Antiguitat        0
Nombre            0
dtype: int64
Valores 'NC':
Any               0
Codi_Districte    0
Nom_Districte     0
Codi_Barri        0
Nom_Barri         0
Seccio_Censal     0
Tipus_Vehicles    0
Antiguitat        0
Nombre            0
dtype: int64
Valores 'No consta':
Any               0
Codi_Districte    0
Nom_Districte     0
Codi_Barri        0
Nom_Barri         0
Seccio_Censal     0
Tipus_Vehicles    0
Antiguitat        0
Nombre            0
dtype: int64


In [31]:
# Determinamos cuantos valores existen de cada variable:

for columna in Antigüedad_tipo_2021_limpio.columns:
    valores_unicos_2021 = Antigüedad_tipo_2021_limpio[columna].value_counts()
    cantidad_valores_unicos_2021 = len(valores_unicos_2021)
    print("Variable:", columna)
    print("Cantidad de valores únicos:", cantidad_valores_unicos_2021)
    print("Nombres de cada valor:")
    print(valores_unicos_2021)
    print()

Variable: Any
Cantidad de valores únicos: 1
Nombres de cada valor:
2021    52741
Name: Any, dtype: int64

Variable: Codi_Districte
Cantidad de valores únicos: 10
Nombres de cada valor:
2     8641
10    7442
7     6034
3     5773
8     5656
5     4883
9     4679
6     4132
4     2794
1     2707
Name: Codi_Districte, dtype: int64

Variable: Nom_Districte
Cantidad de valores únicos: 10
Nombres de cada valor:
Eixample              8641
Sant Martí            7442
Horta-Guinardó        6034
Sants-Montjuïc        5773
Nou Barris            5656
Sarrià-St. Gervasi    4883
Sant Andreu           4679
Gràcia                4132
Les Corts             2794
Ciutat Vella          2707
Name: Nom_Districte, dtype: int64

Variable: Codi_Barri
Cantidad de valores únicos: 73
Nombres de cada valor:
9     1904
60    1854
31    1712
19    1673
6     1648
      ... 
47      94
58      93
12      75
56      58
42      50
Name: Codi_Barri, Length: 73, dtype: int64

Variable: Nom_Barri
Cantidad de valores únicos

#### Datos 2019 - 2020 - 2021

Creamos un nuevo DF con todos los datos obtenidos del 2019, 2020 y 2021, correspondiene al 'Padrón de vehículos. Antigüedad por tipo de vehículo del parque de vehículos de la ciudad de Barcelona': número de vehículos por tipo, antigüedad, distrito y barrio.

In [32]:
# Creamos un DF con el total de los datos para el posterior análisis:

Antigüedad_tipo_TOTAL = pd.concat([Antigüedad_tipo_2019_limpio, Antigüedad_tipo_2020_limpio, Antigüedad_tipo_2021_limpio], axis = 0)

Antigüedad_tipo_TOTAL

Any Codi_Districte Nom_Districte Codi_Barri            Nom_Barri  \
0      2019              1  Ciutat Vella          1             el Raval   
1      2019              1  Ciutat Vella          1             el Raval   
2      2019              1  Ciutat Vella          1             el Raval   
3      2019              1  Ciutat Vella          1             el Raval   
4      2019              1  Ciutat Vella          1             el Raval   
...     ...            ...           ...        ...                  ...   
52877  2021             10    Sant Martí         73  la Verneda i la Pau   
52878  2021             10    Sant Martí         73  la Verneda i la Pau   
52879  2021             10    Sant Martí         73  la Verneda i la Pau   
52880  2021             10    Sant Martí         73  la Verneda i la Pau   
52881  2021             10    Sant Martí         73  la Verneda i la Pau   

      Seccio_Censal   Tipus_Vehicles                   Antiguitat  Nombre  
0                 1         Turismes  Menys d'un any d'antiguitat      19  
1                 1         Turismes                        1 any      10  
2                 1         Turismes                       2 anys      19  
3                 1         Turismes                       3 anys      17  
4                 1         Turismes                       4 anys      19  
...             ...              ...                          ...     ...  
52877           143          Camions               Més de 20 anys       3  
52878           143  Altres vehicles  Menys d'un any d'antiguitat       1  
52879           143  Altres vehicles                       6 anys       1  
52880           143  Altres vehicles               D'11 a 20 anys       2  
52881           143  Altres vehicles               Més de 20 anys       3  

[157448 rows x 9 columns]

In [33]:
# Guardamos el DataFrame en un archivo CSV

Antigüedad_tipo_TOTAL_df = pd.DataFrame(Antigüedad_tipo_TOTAL)
Antigüedad_tipo_TOTAL_df.to_csv('Antigüedad_tipo_TOTAL.csv', index=False)


--------------------------------------------------------------------------------------------------------------------------------

#### Dataframes (distintivo AMB):

- Padrón de vehículos: __Según distintivo ambiental por tipo de vehículo del parque de vehículos de la ciudad de Barcelona__

    - 2019
    - 2020
    - 2021
    - Dataframe total

#### Datos 2019
Carga del archivo '2019_parc_vehicles_distintiu_amb' correspondiene al 'Padrón de vehículos. Según distintivo ambiental por tipo de vehículo del parque de vehículos de la ciudad de Barcelona' en 2019: número de vehículos por tipo, distintivo ambiental, distrito y barrio.

In [34]:
# Carga de archivo '2019_parc_vehicles_distintiu_amb.csv'

pd.read_csv('2019_parc_vehicles_distintiu_amb.csv')

Distintivo_AMB_2019 = pd.read_csv('2019_parc_vehicles_distintiu_amb.csv')

pd.options.display.max_columns = None

Distintivo_AMB_2019

Any Codi_Districte Nom_Districte Codi_Barri  Nom_Barri Seccio_Censal  \
0      2019              1  Ciutat Vella          1   el Raval             1   
1      2019              1  Ciutat Vella          1   el Raval             1   
2      2019              1  Ciutat Vella          1   el Raval             1   
3      2019              1  Ciutat Vella          1   el Raval             1   
4      2019              1  Ciutat Vella          1   el Raval             1   
...     ...            ...           ...        ...        ...           ...   
21429  2019             NC     No consta         NC  No consta            NC   
21430  2019             NC     No consta         NC  No consta            NC   
21431  2019             NC     No consta         NC  No consta            NC   
21432  2019             NC     No consta         NC  No consta            NC   
21433  2019             NC     No consta         NC  No consta            NC   

        Tipus_Vehicles             Desc_Distintiu_Ambiental  Nombre  
0             Turismes        Sense distintiu mediambiental     111  
1             Turismes  Etiqueta 0 - Zero emissions - Blava       5  
2             Turismes                   Etiqueta B - Groga     118  
3             Turismes                   Etiqueta C - Verda     101  
4             Turismes                     Etiqueta E - Eco       9  
...                ...                                  ...     ...  
21429       Furgonetes                   Etiqueta B - Groga       5  
21430       Furgonetes                   Etiqueta C - Verda       8  
21431          Camions        Sense distintiu mediambiental       9  
21432          Camions                            No consta       1  
21433  Altres vehicles        Sense distintiu mediambiental       3  

[21434 rows x 9 columns]

In [35]:
# Estructura del DF

Distintivo_AMB_2019.shape

(21434, 9)

In [36]:
# Obtenemos el total de vehículos con distintivo AMB censados en 2019

total_Distintivo_AMB_2019 = Distintivo_AMB_2019['Nombre'].sum()
print("Total vehículos distintivo AMB Barcelona - 2019:", total_Distintivo_AMB_2019)

Total vehículos distintivo AMB Barcelona - 2019: 824801


In [37]:
# Revisamos en que variables tenemos valores 'null', 'NC' o 'No consta'

# Comprobar valores nulos
valores_nulos_AMB_2019 = Distintivo_AMB_2019.isnull().sum()
print("Valores nulos:")
print(valores_nulos_AMB_2019)

# Comprobar valores 'NC'
valores_nc_AMB_2019 = Distintivo_AMB_2019[Distintivo_AMB_2019 == 'NC'].count()
print("Valores 'NC':")
print(valores_nc_AMB_2019)

# Comprobar valores 'No consta'
valores_no_consta_AMB_2019 = Distintivo_AMB_2019[Distintivo_AMB_2019 == 'No consta'].count()
print("Valores 'No consta':")
print(valores_no_consta_AMB_2019)

Valores nulos:
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64
Valores 'NC':
Any                          0
Codi_Districte              20
Nom_Districte                0
Codi_Barri                  20
Nom_Barri                    0
Seccio_Censal               20
Tipus_Vehicles               0
Desc_Distintiu_Ambiental     0
Nombre                       0
dtype: int64
Valores 'No consta':
Any                            0
Codi_Districte                 0
Nom_Districte                 20
Codi_Barri                     0
Nom_Barri                     20
Seccio_Censal                  0
Tipus_Vehicles                 0
Desc_Distintiu_Ambiental    2292
Nombre                         0
dtype: int64


In [38]:
# Comrpobamos el total de vehículos no clasificados en Distritos ('NC') y el total de vehículos no clasificados por Descripción Distintivo Ambiental ('No consta'):
    # En este caso los valores 'NC' de 'Codi_Districte' coinciden con los valores 'No consta' de 'Nom_Districte',
    ## así como con los valores 'NC' de 'Codi_Barri' y 'No consta' de 'Nom_Barri'.


Total_vehiculos_NC_AMB_2019 = Distintivo_AMB_2019.loc[Distintivo_AMB_2019['Codi_Districte'] == 'NC', 'Nombre'].sum()
print("Total vehículos no clasificados en Distritos - 2019:", Total_vehiculos_NC_2019)

Total_vehiculos_No_Consta_AMB_2019 = Distintivo_AMB_2019.loc[Distintivo_AMB_2019['Desc_Distintiu_Ambiental'] == 'No consta', 'Nombre'].sum()
print("Total vehículos no clasificados por distintivo AMB - 2019:", Total_vehiculos_No_Consta_2019)

Total vehículos no clasificados en Distritos - 2019: 347
Total vehículos no clasificados por distintivo AMB - 2019: 210


In [39]:
# Comprobamos si existe alguna coincidencia entre variables, es decir si alguna fila que contiene variable 'Codi_Districte' = 'NC'
## también contiene variable 'Desc_Distintiu_Ambiental' = 'No consta'

Total_vehiculos_NC_AMB_2019_series = pd.Series([Total_vehiculos_NC_AMB_2019])
Total_vehiculos_No_Consta_AMB_2019_series = pd.Series([Total_vehiculos_No_Consta_AMB_2019])

coincidencia_AMB_2019 = Total_vehiculos_NC_AMB_2019_series.isin(Total_vehiculos_No_Consta_AMB_2019_series).any()

if coincidencia_AMB_2019:
    print("Hay coincidencia en los valores de total vehículos 'Nombre'.")
else:
    print("No hay coincidencia en los valores de total vehículos 'Nombre'.")

No hay coincidencia en los valores de total vehículos 'Nombre'.


#### Observaciones:

Tenemos un total de __824.801__ vehículos censados en Barcelona en 2019, de los cuales __347__ no están clasificados ni por distritos ni por barrios y __210__ no consta su distintivo ambiental. Para el análisis del total del conjunto, teniendo en cuenta que estos 347 vehículos suponen menos de un 0,1% del total del parque automovílistico, los obviamos para no reclasificarlos aleatoriamente a ninguna zona de la ciudad ni con ningún tipo de antigüedad.

Al realziar la comprabación de 'coincindencia' aseguramos que los valores 'NC' de 'Codi_Districte' no coinciden con los valors 'No Consta' de la variable 'Desc_Distintiu_Ambiental', es decir, ningun registro que tenga un valor 'NC' de 'Codi_Districte' tiene una valor 'No Consta' de  'Desc_Distintiu_Ambiental'. Podemos asegurar que un total de __347__ vehículos no clasificados por 'Codi_Districte' y un total de __210__ no clasificados por 'Desc_Distintiu_Ambiental', no coinciden en ningun registro y por tanto pueden obviarse para el análisis del DF.

In [40]:
# Eliminamos la filas que contengan 'Codi_Districte' = 'NC' i las filas que contengan 'Desc_Distintiu_Ambiental' = 'No consta'

Distintivo_AMB_2019_limpio = Distintivo_AMB_2019.drop(Distintivo_AMB_2019[(Distintivo_AMB_2019['Codi_Districte'] == 'NC') | 
        (Distintivo_AMB_2019['Desc_Distintiu_Ambiental'] == 'No consta')].index)

Distintivo_AMB_2019_limpio

Any Codi_Districte Nom_Districte Codi_Barri            Nom_Barri  \
0      2019              1  Ciutat Vella          1             el Raval   
1      2019              1  Ciutat Vella          1             el Raval   
2      2019              1  Ciutat Vella          1             el Raval   
3      2019              1  Ciutat Vella          1             el Raval   
4      2019              1  Ciutat Vella          1             el Raval   
...     ...            ...           ...        ...                  ...   
21409  2019             10    Sant Martí         73  la Verneda i la Pau   
21410  2019             10    Sant Martí         73  la Verneda i la Pau   
21411  2019             10    Sant Martí         73  la Verneda i la Pau   
21412  2019             10    Sant Martí         73  la Verneda i la Pau   
21413  2019             10    Sant Martí         73  la Verneda i la Pau   

      Seccio_Censal   Tipus_Vehicles             Desc_Distintiu_Ambiental  \
0                 1         Turismes        Sense distintiu mediambiental   
1                 1         Turismes  Etiqueta 0 - Zero emissions - Blava   
2                 1         Turismes                   Etiqueta B - Groga   
3                 1         Turismes                   Etiqueta C - Verda   
4                 1         Turismes                     Etiqueta E - Eco   
...             ...              ...                                  ...   
21409           143       Furgonetes                   Etiqueta C - Verda   
21410           143          Camions        Sense distintiu mediambiental   
21411           143          Camions                   Etiqueta B - Groga   
21412           143          Camions                   Etiqueta C - Verda   
21413           143  Altres vehicles        Sense distintiu mediambiental   

       Nombre  
0         111  
1           5  
2         118  
3         101  
4           9  
...       ...  
21409       3  
21410       6  
21411       3  
21412       3  
21413       8  

[19125 rows x 9 columns]

In [41]:
# Comprobamos en el nuevo DF si existe algun valor 'null', 'NC' o 'No consta'

# Comprobar valores nulos
valores_nulos_AMB_2019_limpio = Distintivo_AMB_2019_limpio.isnull().sum()
print("Valores nulos:")
print(valores_nulos_AMB_2019_limpio)

# Comprobar valores 'NC'
valores_nc_AMB_2019_limpio = Distintivo_AMB_2019_limpio[Distintivo_AMB_2019_limpio == 'NC'].count()
print("Valores 'NC':")
print(valores_nc_AMB_2019_limpio)

# Comprobar valores 'No consta'
valores_no_consta_AMB_2019_limpio = Distintivo_AMB_2019_limpio[Distintivo_AMB_2019_limpio == 'No consta'].count()
print("Valores 'No consta':")
print(valores_no_consta_AMB_2019_limpio)

Valores nulos:
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64
Valores 'NC':
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64
Valores 'No consta':
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64


In [43]:
# Determinamos cuantos valores existen de cada variable:

for columna in Distintivo_AMB_2019_limpio.columns:
    valores_unicos_AMB_2019 = Distintivo_AMB_2019_limpio[columna].value_counts()
    cantidad_valores_unicos_AMB_2019 = len(valores_unicos_AMB_2019)
    print("Variable:", columna)
    print("Cantidad de valores únicos:", cantidad_valores_unicos_AMB_2019)
    print("Nombres de cada valor:")
    print(valores_unicos_AMB_2019)
    print()

Variable: Any
Cantidad de valores únicos: 1
Nombres de cada valor:
2019    19125
Name: Any, dtype: int64

Variable: Codi_Districte
Cantidad de valores únicos: 10
Nombres de cada valor:
2     3159
10    2633
7     2171
3     2069
8     2002
5     1835
9     1690
6     1539
4     1038
1      989
Name: Codi_Districte, dtype: int64

Variable: Nom_Districte
Cantidad de valores únicos: 10
Nombres de cada valor:
Eixample              3159
Sant Martí            2633
Horta-Guinardó        2171
Sants-Montjuïc        2069
Nou Barris            2002
Sarrià-St. Gervasi    1835
Sant Andreu           1690
Gràcia                1539
Les Corts             1038
Ciutat Vella           989
Name: Nom_Districte, dtype: int64

Variable: Codi_Barri
Cantidad de valores únicos: 73
Nombres de cada valor:
9     715
60    694
31    640
19    632
6     601
     ... 
47     37
58     31
12     28
56     20
42     15
Name: Codi_Barri, Length: 73, dtype: int64

Variable: Nom_Barri
Cantidad de valores únicos: 73
Nombre

------------------------------------------------------------------------------------------------------------------------------

#### Datos 2020
Carga del archivo '2020_parc_vehicles_distintiu_amb' correspondiene al 'Padrón de vehículos. Según distintivo ambiental por tipo de vehículo del parque de vehículos de la ciudad de Barcelona' en 2020: número de vehículos por tipo, distintivo ambiental, distrito y barrio.

In [44]:
# Carga de archivo '2020_parc_vehicles_distintiu_amb.csv'

pd.read_csv('2020_parc_vehicles_distintiu_amb.csv')

Distintivo_AMB_2020 = pd.read_csv('2020_parc_vehicles_distintiu_amb.csv')

pd.options.display.max_columns = None

Distintivo_AMB_2020

Any Codi_Districte Nom_Districte Codi_Barri  Nom_Barri Seccio_Censal  \
0      2020              1  Ciutat Vella          1   el Raval             1   
1      2020              1  Ciutat Vella          1   el Raval             1   
2      2020              1  Ciutat Vella          1   el Raval             1   
3      2020              1  Ciutat Vella          1   el Raval             1   
4      2020              1  Ciutat Vella          1   el Raval             1   
...     ...            ...           ...        ...        ...           ...   
22187  2020             NC     No consta         NC  No consta            NC   
22188  2020             NC     No consta         NC  No consta            NC   
22189  2020             NC     No consta         NC  No consta            NC   
22190  2020             NC     No consta         NC  No consta            NC   
22191  2020             NC     No consta         NC  No consta            NC   

        Tipus_Vehicles             Desc_Distintiu_Ambiental  Nombre  
0             Turismes        Sense distintiu mediambiental      95  
1             Turismes  Etiqueta 0 - Zero emissions - Blava       5  
2             Turismes                   Etiqueta B - Groga     118  
3             Turismes                   Etiqueta C - Verda     118  
4             Turismes                     Etiqueta E - Eco       9  
...                ...                                  ...     ...  
22187       Furgonetes                   Etiqueta C - Verda       3  
22188          Camions        Sense distintiu mediambiental       9  
22189          Camions                   Etiqueta B - Groga       1  
22190          Camions                            No consta       1  
22191  Altres vehicles        Sense distintiu mediambiental       5  

[22192 rows x 9 columns]

In [45]:
# Estructura del DF

Distintivo_AMB_2020.shape

(22192, 9)

In [46]:
# Obtenemos el total de vehículos con distintivo AMB censados en 2020

total_Distintivo_AMB_2020 = Distintivo_AMB_2020['Nombre'].sum()
print("Total vehículos distintivo AMB Barcelona - 2020:", total_Distintivo_AMB_2020)

Total vehículos distintivo AMB Barcelona - 2020: 822211


In [47]:
# Revisamos en que variables tenemos valores 'null', 'NC' o 'No consta'

# Comprobar valores nulos
valores_nulos_AMB_2020 = Distintivo_AMB_2020.isnull().sum()
print("Valores nulos:")
print(valores_nulos_AMB_2020)

# Comprobar valores 'NC'
valores_nc_AMB_2020 = Distintivo_AMB_2020[Distintivo_AMB_2020 == 'NC'].count()
print("Valores 'NC':")
print(valores_nc_AMB_2020)

# Comprobar valores 'No consta'
valores_no_consta_AMB_2020 = Distintivo_AMB_2020[Distintivo_AMB_2020 == 'No consta'].count()
print("Valores 'No consta':")
print(valores_no_consta_AMB_2020)

Valores nulos:
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64
Valores 'NC':
Any                          0
Codi_Districte              20
Nom_Districte                0
Codi_Barri                  20
Nom_Barri                    0
Seccio_Censal               20
Tipus_Vehicles               0
Desc_Distintiu_Ambiental     0
Nombre                       0
dtype: int64
Valores 'No consta':
Any                            0
Codi_Districte                 0
Nom_Districte                 20
Codi_Barri                     0
Nom_Barri                     20
Seccio_Censal                  0
Tipus_Vehicles                 0
Desc_Distintiu_Ambiental    2254
Nombre                         0
dtype: int64


In [48]:
# Comrpobamos el total de vehículos no clasificados en Distritos ('NC') y el total de vehículos no clasificados por Descripción Distintivo Ambiental ('No consta'):
    # En este caso los valores 'NC' de 'Codi_Districte' coinciden con los valores 'No consta' de 'Nom_Districte',
    ## así como con los valores 'NC' de 'Codi_Barri' y 'No consta' de 'Nom_Barri'.


Total_vehiculos_NC_AMB_2020 = Distintivo_AMB_2020.loc[Distintivo_AMB_2020['Codi_Districte'] == 'NC', 'Nombre'].sum()
print("Total vehículos no clasificados en Distritos - 2020:", Total_vehiculos_NC_2020)

Total_vehiculos_No_Consta_AMB_2020 = Distintivo_AMB_2020.loc[Distintivo_AMB_2020['Desc_Distintiu_Ambiental'] == 'No consta', 'Nombre'].sum()
print("Total vehículos no clasificados por distintivo AMB - 2020:", Total_vehiculos_No_Consta_2020)

Total vehículos no clasificados en Distritos - 2020: 415
Total vehículos no clasificados por distintivo AMB - 2020: 158


In [49]:
# Comprobamos si existe alguna coincidencia entre variables, es decir si alguna fila que contiene variable 'Codi_Districte' = 'NC'
## también contiene variable 'Desc_Distintiu_Ambiental' = 'No consta'

Total_vehiculos_NC_AMB_2020_series = pd.Series([Total_vehiculos_NC_AMB_2020])
Total_vehiculos_No_Consta_AMB_2020_series = pd.Series([Total_vehiculos_No_Consta_AMB_2020])

coincidencia_AMB_2020 = Total_vehiculos_NC_AMB_2020_series.isin(Total_vehiculos_No_Consta_AMB_2020_series).any()

if coincidencia_AMB_2020:
    print("Hay coincidencia en los valores de total vehículos 'Nombre'.")
else:
    print("No hay coincidencia en los valores de total vehículos 'Nombre'.")

No hay coincidencia en los valores de total vehículos 'Nombre'.


#### Observaciones:

Tenemos un total de __822.211__ vehículos censados en Barcelona en 2020, de los cuales __415__ no están clasificados ni por distritos ni por barrios y __158__ no consta su distintivo ambiental. Para el análisis del total del conjunto, teniendo en cuenta que estos 573 vehículos suponen menos de un 0,1% del total del parque automovílistico, los obviamos para no reclasificarlos aleatoriamente a ninguna zona de la ciudad ni con ningún tipo de antigüedad.

Al realziar la comprabación de 'coincindencia' aseguramos que los valores 'NC' de 'Codi_Districte' no coinciden con los valors 'No Consta' de la variable 'Desc_Distintiu_Ambiental', es decir, ningun registro que tenga un valor 'NC' de 'Codi_Districte' tiene una valor 'No Consta' de  'Desc_Distintiu_Ambiental'. Podemos asegurar que un total de __415__ vehículos no clasificados por 'Codi_Districte' y un total de __158__ no clasificados por 'Desc_Distintiu_Ambiental', no coinciden en ningun registro y por tanto pueden obviarse para el análisis del DF.

In [50]:
# Eliminamos la filas que contengan 'Codi_Districte' = 'NC' i las filas que contengan 'Desc_Distintiu_Ambiental' = 'No consta'

Distintivo_AMB_2020_limpio = Distintivo_AMB_2020.drop(Distintivo_AMB_2020[(Distintivo_AMB_2020['Codi_Districte'] == 'NC') | 
        (Distintivo_AMB_2020['Desc_Distintiu_Ambiental'] == 'No consta')].index)

Distintivo_AMB_2020_limpio

Any Codi_Districte Nom_Districte Codi_Barri            Nom_Barri  \
0      2020              1  Ciutat Vella          1             el Raval   
1      2020              1  Ciutat Vella          1             el Raval   
2      2020              1  Ciutat Vella          1             el Raval   
3      2020              1  Ciutat Vella          1             el Raval   
4      2020              1  Ciutat Vella          1             el Raval   
...     ...            ...           ...        ...                  ...   
22167  2020             10    Sant Martí         73  la Verneda i la Pau   
22168  2020             10    Sant Martí         73  la Verneda i la Pau   
22169  2020             10    Sant Martí         73  la Verneda i la Pau   
22170  2020             10    Sant Martí         73  la Verneda i la Pau   
22171  2020             10    Sant Martí         73  la Verneda i la Pau   

      Seccio_Censal   Tipus_Vehicles             Desc_Distintiu_Ambiental  \
0                 1         Turismes        Sense distintiu mediambiental   
1                 1         Turismes  Etiqueta 0 - Zero emissions - Blava   
2                 1         Turismes                   Etiqueta B - Groga   
3                 1         Turismes                   Etiqueta C - Verda   
4                 1         Turismes                     Etiqueta E - Eco   
...             ...              ...                                  ...   
22167           143       Furgonetes                   Etiqueta C - Verda   
22168           143          Camions        Sense distintiu mediambiental   
22169           143          Camions                   Etiqueta B - Groga   
22170           143          Camions                   Etiqueta C - Verda   
22171           143  Altres vehicles        Sense distintiu mediambiental   

       Nombre  
0          95  
1           5  
2         118  
3         118  
4           9  
...       ...  
22167       5  
22168       7  
22169       5  
22170       3  
22171       8  

[19921 rows x 9 columns]

In [51]:
# Comprobamos en el nuevo DF si existe algun valor 'null', 'NC' o 'No consta'

# Comprobar valores nulos
valores_nulos_AMB_2020_limpio = Distintivo_AMB_2020_limpio.isnull().sum()
print("Valores nulos:")
print(valores_nulos_AMB_2020_limpio)

# Comprobar valores 'NC'
valores_nc_AMB_2020_limpio = Distintivo_AMB_2020_limpio[Distintivo_AMB_2020_limpio == 'NC'].count()
print("Valores 'NC':")
print(valores_nc_AMB_2020_limpio)

# Comprobar valores 'No consta'
valores_no_consta_AMB_2020_limpio = Distintivo_AMB_2020_limpio[Distintivo_AMB_2020_limpio == 'No consta'].count()
print("Valores 'No consta':")
print(valores_no_consta_AMB_2020_limpio)

Valores nulos:
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64
Valores 'NC':
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64
Valores 'No consta':
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64


In [53]:
# Determinamos cuantos valores existen de cada variable:

for columna in Distintivo_AMB_2020_limpio.columns:
    valores_unicos_AMB_2020 = Distintivo_AMB_2020_limpio[columna].value_counts()
    cantidad_valores_unicos_AMB_2020 = len(valores_unicos_AMB_2020)
    print("Variable:", columna)
    print("Cantidad de valores únicos:", cantidad_valores_unicos_AMB_2020)
    print("Nombres de cada valor:")
    print(valores_unicos_AMB_2020)
    print()

Variable: Any
Cantidad de valores únicos: 1
Nombres de cada valor:
2020    19921
Name: Any, dtype: int64

Variable: Codi_Districte
Cantidad de valores únicos: 10
Nombres de cada valor:
2     3293
10    2748
7     2261
3     2163
8     2101
5     1894
9     1754
6     1613
4     1080
1     1014
Name: Codi_Districte, dtype: int64

Variable: Nom_Districte
Cantidad de valores únicos: 10
Nombres de cada valor:
Eixample              3293
Sant Martí            2748
Horta-Guinardó        2261
Sants-Montjuïc        2163
Nou Barris            2101
Sarrià-St. Gervasi    1894
Sant Andreu           1754
Gràcia                1613
Les Corts             1080
Ciutat Vella          1014
Name: Nom_Districte, dtype: int64

Variable: Codi_Barri
Cantidad de valores únicos: 73
Nombres de cada valor:
9     742
60    719
31    672
19    657
6     628
     ... 
47     36
58     33
12     28
56     22
42     17
Name: Codi_Barri, Length: 73, dtype: int64

Variable: Nom_Barri
Cantidad de valores únicos: 73
Nombre

-------------------------------------------------------------------------------------------------------------------------------

#### Datos 2021
Carga del archivo '2021_parc_vehicles_distintiu_amb' correspondiene al 'Padrón de vehículos. Según distintivo ambiental por tipo de vehículo del parque de vehículos de la ciudad de Barcelona' en 2021: número de vehículos por tipo, distintivo ambiental, distrito y barrio.

In [54]:
# Carga de archivo '2021_parc_vehicles_distintiu_amb.csv'

pd.read_csv('2021_parc_vehicles_distintiu_amb.csv')

Distintivo_AMB_2021 = pd.read_csv('2021_parc_vehicles_distintiu_amb.csv')

pd.options.display.max_columns = None

Distintivo_AMB_2021

Any Codi_Districte Nom_Districte Codi_Barri  Nom_Barri Seccio_Censal  \
0      2021              1  Ciutat Vella          1   el Raval             1   
1      2021              1  Ciutat Vella          1   el Raval             1   
2      2021              1  Ciutat Vella          1   el Raval             1   
3      2021              1  Ciutat Vella          1   el Raval             1   
4      2021              1  Ciutat Vella          1   el Raval             1   
...     ...            ...           ...        ...        ...           ...   
23444  2021             NC     No consta         NC  No consta            NC   
23445  2021             NC     No consta         NC  No consta            NC   
23446  2021             NC     No consta         NC  No consta            NC   
23447  2021             NC     No consta         NC  No consta            NC   
23448  2021             NC     No consta         NC  No consta            NC   

        Tipus_Vehicles             Desc_Distintiu_Ambiental  Nombre  
0             Turismes        Sense distintiu mediambiental      83  
1             Turismes  Etiqueta 0 - Zero emissions - Blava       7  
2             Turismes                   Etiqueta B - Groga     107  
3             Turismes                   Etiqueta C - Verda     122  
4             Turismes                     Etiqueta E - Eco      12  
...                ...                                  ...     ...  
23444       Furgonetes                     Etiqueta E - Eco       1  
23445          Camions        Sense distintiu mediambiental       7  
23446          Camions                   Etiqueta B - Groga       1  
23447          Camions                            No consta       1  
23448  Altres vehicles        Sense distintiu mediambiental       4  

[23449 rows x 9 columns]

In [55]:
# Estructura del DF

Distintivo_AMB_2021.shape

(23449, 9)

In [56]:
# Obtenemos el total de vehículos con distintivo AMB censados en 2021

total_Distintivo_AMB_2021 = Distintivo_AMB_2021['Nombre'].sum()
print("Total vehículos distintivo AMB Barcelona - 2021:", total_Distintivo_AMB_2021)

Total vehículos distintivo AMB Barcelona - 2021: 811673


In [57]:
# Revisamos en que variables tenemos valores 'null', 'NC' o 'No consta'

# Comprobar valores nulos
valores_nulos_AMB_2021 = Distintivo_AMB_2021.isnull().sum()
print("Valores nulos:")
print(valores_nulos_AMB_2021)

# Comprobar valores 'NC'
valores_nc_AMB_2021 = Distintivo_AMB_2020[Distintivo_AMB_2021 == 'NC'].count()
print("Valores 'NC':")
print(valores_nc_AMB_2021)

# Comprobar valores 'No consta'
valores_no_consta_AMB_2021 = Distintivo_AMB_2021[Distintivo_AMB_2021 == 'No consta'].count()
print("Valores 'No consta':")
print(valores_no_consta_AMB_2021)

Valores nulos:
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64
Valores 'NC':
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64
Valores 'No consta':
Any                            0
Codi_Districte                 0
Nom_Districte                 22
Codi_Barri                     0
Nom_Barri                     22
Seccio_Censal                  0
Tipus_Vehicles                 0
Desc_Distintiu_Ambiental    2683
Nombre                         0
dtype: int64


In [58]:
# Comrpobamos el total de vehículos no clasificados en Distritos ('NC') y el total de vehículos no clasificados por Descripción Distintivo Ambiental ('No consta'):
    # En este caso los valores 'NC' de 'Codi_Districte' coinciden con los valores 'No consta' de 'Nom_Districte',
    ## así como con los valores 'NC' de 'Codi_Barri' y 'No consta' de 'Nom_Barri'.


Total_vehiculos_NC_AMB_2021 = Distintivo_AMB_2021.loc[Distintivo_AMB_2021['Codi_Districte'] == 'NC', 'Nombre'].sum()
print("Total vehículos no clasificados en Distritos - 2021:", Total_vehiculos_NC_2021)

Total_vehiculos_No_Consta_AMB_2021 = Distintivo_AMB_2021.loc[Distintivo_AMB_2021['Desc_Distintiu_Ambiental'] == 'No consta', 'Nombre'].sum()
print("Total vehículos no clasificados por distintivo AMB - 2021:", Total_vehiculos_No_Consta_2021)

Total vehículos no clasificados en Distritos - 2021: 394
Total vehículos no clasificados por distintivo AMB - 2021: 179


In [59]:
# Comprobamos si existe alguna coincidencia entre variables, es decir si alguna fila que contiene variable 'Codi_Districte' = 'NC'
## también contiene variable 'Desc_Distintiu_Ambiental' = 'No consta'

Total_vehiculos_NC_AMB_2021_series = pd.Series([Total_vehiculos_NC_AMB_2021])
Total_vehiculos_No_Consta_AMB_2021_series = pd.Series([Total_vehiculos_No_Consta_AMB_2021])

coincidencia_AMB_2021 = Total_vehiculos_NC_AMB_2021_series.isin(Total_vehiculos_No_Consta_AMB_2021_series).any()

if coincidencia_AMB_2021:
    print("Hay coincidencia en los valores de total vehículos 'Nombre'.")
else:
    print("No hay coincidencia en los valores de total vehículos 'Nombre'.")

No hay coincidencia en los valores de total vehículos 'Nombre'.


#### Observaciones:

Tenemos un total de __811.673__ vehículos censados en Barcelona en 2020, de los cuales __394__ no están clasificados ni por distritos ni por barrios y __179__ no consta su distintivo ambiental. Para el análisis del total del conjunto, teniendo en cuenta que estos 573 vehículos suponen menos de un 0,1% del total del parque automovílistico, los obviamos para no reclasificarlos aleatoriamente a ninguna zona de la ciudad ni con ningún tipo de antigüedad.

Al realziar la comprabación de 'coincindencia' aseguramos que los valores 'NC' de 'Codi_Districte' no coinciden con los valors 'No Consta' de la variable 'Desc_Distintiu_Ambiental', es decir, ningun registro que tenga un valor 'NC' de 'Codi_Districte' tiene una valor 'No Consta' de  'Desc_Distintiu_Ambiental'. Podemos asegurar que un total de __394__ vehículos no clasificados por 'Codi_Districte' y un total de __179__ no clasificados por 'Desc_Distintiu_Ambiental', no coinciden en ningun registro y por tanto pueden obviarse para el análisis del DF.

In [60]:
# Eliminamos la filas que contengan 'Codi_Districte' = 'NC' i las filas que contengan 'Desc_Distintiu_Ambiental' = 'No consta'

Distintivo_AMB_2021_limpio = Distintivo_AMB_2021.drop(Distintivo_AMB_2021[(Distintivo_AMB_2021['Codi_Districte'] == 'NC') | 
        (Distintivo_AMB_2021['Desc_Distintiu_Ambiental'] == 'No consta')].index)

Distintivo_AMB_2021_limpio

Any Codi_Districte Nom_Districte Codi_Barri            Nom_Barri  \
0      2021              1  Ciutat Vella          1             el Raval   
1      2021              1  Ciutat Vella          1             el Raval   
2      2021              1  Ciutat Vella          1             el Raval   
3      2021              1  Ciutat Vella          1             el Raval   
4      2021              1  Ciutat Vella          1             el Raval   
...     ...            ...           ...        ...                  ...   
23422  2021             10    Sant Martí         73  la Verneda i la Pau   
23423  2021             10    Sant Martí         73  la Verneda i la Pau   
23424  2021             10    Sant Martí         73  la Verneda i la Pau   
23425  2021             10    Sant Martí         73  la Verneda i la Pau   
23426  2021             10    Sant Martí         73  la Verneda i la Pau   

      Seccio_Censal   Tipus_Vehicles             Desc_Distintiu_Ambiental  \
0                 1         Turismes        Sense distintiu mediambiental   
1                 1         Turismes  Etiqueta 0 - Zero emissions - Blava   
2                 1         Turismes                   Etiqueta B - Groga   
3                 1         Turismes                   Etiqueta C - Verda   
4                 1         Turismes                     Etiqueta E - Eco   
...             ...              ...                                  ...   
23422           143          Camions        Sense distintiu mediambiental   
23423           143          Camions                   Etiqueta B - Groga   
23424           143          Camions                   Etiqueta C - Verda   
23425           143  Altres vehicles        Sense distintiu mediambiental   
23426           143  Altres vehicles                   Etiqueta C - Verda   

       Nombre  
0          83  
1           7  
2         107  
3         122  
4          12  
...       ...  
23422       6  
23423       6  
23424       4  
23425       6  
23426       1  

[20747 rows x 9 columns]

In [61]:
# Comprobamos en el nuevo DF si existe algun valor 'null', 'NC' o 'No consta'

# Comprobar valores nulos
valores_nulos_AMB_2021_limpio = Distintivo_AMB_2021_limpio.isnull().sum()
print("Valores nulos:")
print(valores_nulos_AMB_2021_limpio)

# Comprobar valores 'NC'
valores_nc_AMB_2021_limpio = Distintivo_AMB_2021_limpio[Distintivo_AMB_2021_limpio == 'NC'].count()
print("Valores 'NC':")
print(valores_nc_AMB_2021_limpio)

# Comprobar valores 'No consta'
valores_no_consta_AMB_2021_limpio = Distintivo_AMB_2021_limpio[Distintivo_AMB_2021_limpio == 'No consta'].count()
print("Valores 'No consta':")
print(valores_no_consta_AMB_2021_limpio)

Valores nulos:
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64
Valores 'NC':
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64
Valores 'No consta':
Any                         0
Codi_Districte              0
Nom_Districte               0
Codi_Barri                  0
Nom_Barri                   0
Seccio_Censal               0
Tipus_Vehicles              0
Desc_Distintiu_Ambiental    0
Nombre                      0
dtype: int64


In [66]:
# Determinamos cuantos valores existen de cada variable:

for columna in Distintivo_AMB_2021_limpio.columns:
    valores_unicos_AMB_2021 = Distintivo_AMB_2021_limpio[columna].value_counts()
    cantidad_valores_unicos_AMB_2021 = len(valores_unicos_AMB_202)
    print("Variable:", columna)
    print("Cantidad de valores únicos:", cantidad_valores_unicos_AMB_2021)
    print("Nombres de cada valor:")
    print(valores_unicos_AMB_2021)
    print()

NameError: name 'valores_unicos_AMB_202' is not defined

#### Datos 2019 - 2020 - 2021

Creamos un nuevo DF con todos los datos obtenidos del 2019, 2020 y 2021, correspondiene al 'PPadrón de vehículos. Según distintivo ambiental por tipo de vehículo del parque de vehículos de la ciudad de Barcelona': número de vehículos por tipo, distintivo ambiental, distrito y barrio.

In [67]:
# Creamos un DF con el total de los datos para el posterior análisis:

Distintivo_AMB_TOTAL = pd.concat([Distintivo_AMB_2019_limpio, Distintivo_AMB_2020_limpio, Distintivo_AMB_2021_limpio], axis = 0)

Distintivo_AMB_TOTAL

Any Codi_Districte Nom_Districte Codi_Barri            Nom_Barri  \
0      2019              1  Ciutat Vella          1             el Raval   
1      2019              1  Ciutat Vella          1             el Raval   
2      2019              1  Ciutat Vella          1             el Raval   
3      2019              1  Ciutat Vella          1             el Raval   
4      2019              1  Ciutat Vella          1             el Raval   
...     ...            ...           ...        ...                  ...   
23422  2021             10    Sant Martí         73  la Verneda i la Pau   
23423  2021             10    Sant Martí         73  la Verneda i la Pau   
23424  2021             10    Sant Martí         73  la Verneda i la Pau   
23425  2021             10    Sant Martí         73  la Verneda i la Pau   
23426  2021             10    Sant Martí         73  la Verneda i la Pau   

      Seccio_Censal   Tipus_Vehicles             Desc_Distintiu_Ambiental  \
0                 1         Turismes        Sense distintiu mediambiental   
1                 1         Turismes  Etiqueta 0 - Zero emissions - Blava   
2                 1         Turismes                   Etiqueta B - Groga   
3                 1         Turismes                   Etiqueta C - Verda   
4                 1         Turismes                     Etiqueta E - Eco   
...             ...              ...                                  ...   
23422           143          Camions        Sense distintiu mediambiental   
23423           143          Camions                   Etiqueta B - Groga   
23424           143          Camions                   Etiqueta C - Verda   
23425           143  Altres vehicles        Sense distintiu mediambiental   
23426           143  Altres vehicles                   Etiqueta C - Verda   

       Nombre  
0         111  
1           5  
2         118  
3         101  
4           9  
...       ...  
23422       6  
23423       6  
23424       4  
23425       6  
23426       1  

[59793 rows x 9 columns]

In [68]:
# Guardamos el DataFrame en un archivo CSV

Distintivo_AMB_TOTAL_df = pd.DataFrame(Distintivo_AMB_TOTAL)
Distintivo_AMB_TOTAL_df.to_csv('Distintivo_AMB_TOTAL.csv', index=False)